Task 06-3, 08-1, 08-2, 09-1, 09-2, 10-1

anatolimironau opened the pull request:
- implemented tasks; 
  - minor refactoring;
  

Tested: Locally via Chrome

Commit:
Task 08-forms-fetch-api-part-2

- init;

Tested:
Locally via Chrome

anatolimironau committed:
Task 08-forms-fetch-api-part-2

- improved task;

Tested:
Locally via Chrome

Commit:
Task 08-forms-fetch-api-part-2

- improved task;

Tested:
Locally via Chrome

anatolimironau committed:
Task 08-forms-fetch-api-part-2

- improved task '1-product-form-v1';

Tested:
Locally via Chrome

Commit:
Task 08-forms-fetch-api-part-2

- improved task '1-product-form-v1';

Tested:
Locally via Chrome

Commit:
Task 08-forms-fetch-api-part-2

- implemented task '1-product-form-v1';

Tested:
Locally via Chrome

Commit:
Task 08-forms-fetch-api-part-2

- implemented task '1-product-form-v1';

Tested:
Locally via Chrome

jsru commented:
Добавляю преподавателя (@dosandk) для код-ревью.

Commit:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20210427_anatoly-mironov into master

Commit:
Task 08-forms-fetch-api-part-2/2-range-picker

- added task;

Tested:
Locally via Chrome

dosandk commented on 08-forms-fetch-api-part-2/1-product-form-v1/index.js:
```diff
@@ -5,11 +5,304 @@ const IMGUR_CLIENT_ID = '28aaa2e823b03b1';
+    if (this.productId) {
+      [...this.formFields.strings, ...this.formFields.numbers].forEach(field => {
+        this.subElements['productForm'].querySelector(`#${field}`).value = this.productData[field];
+        // todo: @dosandk, не понимаю, почему при использовании строки ниже не работают тесты. Но при этом в браузере всё отрабатывает корректно. Подскажите, пожалуйста
```
@anatolimironau 
1е самое важное. Что делает эта строка?

Ну и ряд дополнительных вопросов:
* какой тест/тесты не работают? Все или конкретные?
* Есть ли какие-то ошибки?
* Как дебажили тесты?

anatolimironau replied:
Ответил в ЛС Slack

anatolimironau committed:
Task 08-forms-fetch-api-part-2/2-range-picker

- added dispatch-event();

Tested:
Locally via Chrome

anatolimironau committed:
Task 08-forms-fetch-api-part-2/2-range-picker

- solution refactoring;

Tested:
Locally via Chrome

anatolimironau committed:
Merge remote-tracking branch 'origin/master'

anatolimironau committed:
Implemented task 08-forms-fetch-api-part-2

- fixed tests for 2-range-picker

Tested: Locally via Chrome

anatolimironau committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20210427_anatoly-mironov

Commit:
Task 08-forms-fetch-api-part-2

- removed obsolete code;

Tested:
Locally via Chrome

anatolimironau committed:
Merge remote-tracking branch 'origin/master'

anatolimironau committed:
Implemented task 09-tests-for-frontend-apps

- implemented "2-sortable-list"

Tested: Locally via Chrome

anatolimironau committed:
Implemented task 09-tests-for-frontend-apps

- bugfixing

Tested: Locally via Chrome

anatolimironau committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20210427_anatoly-mironov

anatolimironau committed:
Implemented task 09-tests-for-frontend-apps

- added task 1-product-form-v2
- minor refactoring;

Tested: Locally via Chrome

anatolimironau committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20210427_anatoly-mironov

anatolimironau committed:
Implemented task 10-routes-browser-history-api

- added '1-dashboard-page';
- minor refactoring;

Tested: Locally via Chrome

anatolimironau committed:
Implemented task 10-routes-browser-history-api

- minor refactoring;

Tested: Locally via Chrome

anatolimironau committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20210427_anatoly-mironov

anatolimironau committed:
Task 06-events-practice

- added '3-double-slider' solution;
- minor refactoring;

Tested: Locally via Chrome
